### PR TITLE
Add some explanation about the predeploy Hook code

### PR DESCRIPTION
**Pull request by jshackell-sfdc**
### What does this PR do?

Adds a section to README that attempts to explain some of the Typescript code in the sample predeploy hook.  

@amphro reviewed this a bit, but I think it needs another pair of eyes to make sure I described stuff clearly.  I sorta made up a good portion of it :).   And maybe there's more useful info that can be added?  Feel free to change as needed.  I will link to this from the Hooks docs in the SFDX Plug-In Dev Guide. 

Note that I don't have write access to this repo, which is why i created a fork then PR.

### CODE REVIEW

File: README.md
Comment (amphro) [inline README.md]: Can we have this at the top?

**Review by amphro — CHANGES_REQUESTED** [review]

File: README.md
Comment (jshackell-sfdc) [reply README.md]: I renamed my new section "About the Predeploy Hook TypeScript Code" (which better describes what the section is really about) and moved it up. 

I don't have write access to the repo so I can't merge the change...

**Review by amphro — APPROVED** [review]

**Pull request merged**
